druntime/src/rt/lifetime.d

```diff
@@ -32,6 +32,161 @@ private
 
 }
 
+// NOTE: these are going to be migrated to the GC soon. They are here to allow
+// for better review for now.
+private void[] getArrayUsed(void *ptr, bool atomic = false) nothrow
+{
+    // lookup the block info, using the cache if possible.
+    auto bic = atomic ? null : __getBlkInfo(ptr);
+    auto info = bic ? *bic : GC.query(ptr);
+
+    if (!(info.attr & BlkAttr.APPENDABLE))
+        // not appendable
+        return null;
+
+    assert(info.base); // sanity check.
+    if (!bic && !atomic)
+        // cache the lookup for next time
+        __insertBlkInfoCache(info, null);
+
+    auto usedSize = atomic ? __arrayAllocLengthAtomic(info) : __arrayAllocLength(info);
+    return __arrayStart(info)[0 .. usedSize];
+}
+
+private bool expandArrayUsed(void[] slice, size_t newUsed, bool atomic = false) nothrow
+{
+    if (newUsed < slice.length)
+        // cannot "expand" by shrinking.
+        return false;
+
+    // lookup the block info, using the cache if possible
+    auto bic = atomic ? null : __getBlkInfo(slice.ptr);
+    auto info = bic ? *bic : GC.query(slice.ptr);
+
+    if (!(info.attr & BlkAttr.APPENDABLE))
+        // not appendable
+        return false;
+
+    assert(info.base); // sanity check.
+
+    immutable offset = slice.ptr - __arrayStart(info);
+    newUsed += offset;
+    auto existingUsed = slice.length + offset;
+
+    size_t typeInfoSize = (info.attr & BlkAttr.STRUCTFINAL) ? size_t.sizeof : 0;
+    if (__setArrayAllocLengthImpl(info, offset + newUsed, atomic, existingUsed, typeInfoSize))
+    {
+        // could expand without extending
+        if (!bic && !atomic)
+            // cache the lookup for next time
+            __insertBlkInfoCache(info, null);
+        return true;
+    }
+
+    // if we got here, just setting the used size did not work.
+    if (info.size < PAGESIZE)
+        // nothing else we can do
+        return false;
+
+    // try extending the block into subsequent pages.
+    immutable requiredExtension = newUsed - info.size - LARGEPAD;
+    auto extendedSize = GC.extend(info.base, requiredExtension, requiredExtension);
+    if (extendedSize == 0)
+        // could not extend, can't satisfy the request
+        return false;
+
+    info.size = extendedSize;
+    if (bic)
+        *bic = info;
+    else if (!atomic)
+        __insertBlkInfoCache(info, null);
+
+    // this should always work.
+    return __setArrayAllocLengthImpl(info, newUsed, atomic, existingUsed, typeInfoSize);
+}
+
+private bool shrinkArrayUsed(void[] slice, size_t existingUsed, bool atomic = false) nothrow
+{
+    if(existingUsed < slice.length)
```

Comment:
space between `if` and `(`

Reply:
Surprised the style checker passed this?